packages/frontend/app/components/LiquidityConfirmDialog.tsx

```diff
@@ -0,0 +1,70 @@
+import { Dialog } from '@headlessui/react'
+import { Form } from '@remix-run/react'
+import { XIcon } from '~/components/icons'
+import { Button } from '~/components/ui'
+
+type LiquidityConfirmDialogProps = {
+  title: string
+  onClose: () => void
+  type: 'Deposit' | 'Withdraw'
+}
+
+export const LiquidityConfirmDialog = ({
+  title,
+  onClose,
+  type
+}: LiquidityConfirmDialogProps) => {
+  return (
+    <Dialog as='div' className='relative z-10' onClose={onClose} open={true}>
+      <div className='fixed inset-0 bg-tealish/30 bg-opacity-75 transition-opacity' />
+      <div className='fixed inset-0 z-10 overflow-y-auto'>
+        <div className='flex min-h-full items-center justify-center p-4 text-center'>
+          <Dialog.Panel className='relative transform overflow-hidden rounded-lg max-w-full transition-all bg-white px-4 pb-4 pt-5 text-left shadow-xl w-full sm:max-w-lg'>
+            <div className='absolute right-0 top-0 pr-4 pt-4'>
+              <button
+                type='button'
+                className='text-gray-400 hover:text-gray-500 focus:outline-none'
+                onClick={onClose}
+              >
+                <span className='sr-only'>Close</span>
+                <XIcon className='h-8 w-8' aria-hidden='true' />
+              </button>
+            </div>
+            <div>
+              <Dialog.Title
+                as='h3'
+                className='font-semibold leading-6 text-lg text-center'
+              >
+                {title}
+              </Dialog.Title>
+              <div className='mt-2'>
+                <p className='m-6 text-center'>
+                  Are you sure you want to {type.toLowerCase()} this amount?
```

Comment:
should we pass in the actual amount into this confirmation?

Reply:
Added using the outlet context (https://remix.run/docs/en/main/components/outlet#context). Wasn't aware of that feature but was happy to find it (compared to using a query variable, react context, local storage).